app/src/main/java/ghcharactertracker/com/ghcharactertracker/DBHandler.java

```diff
@@ -0,0 +1,159 @@
+package ghcharactertracker.com.ghcharactertracker;
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+/**
+ * Created by jesse.mailhot on 3/20/2018.
+ */
+
+public class DBHandler {
+    private static DBHandler dbHandler;
+    private SQLiteDatabase db;
+
+    private DBHandler(Context context) {
+        db = context.openOrCreateDatabase("DATABASE", Context.MODE_PRIVATE, null);
+
+        db.execSQL(PlayerContract.PlayerEntry.OPEN_OR_CREATE_CHARACTERS);
+        db.execSQL(PlayerContract.ScenarioEntry.OPEN_OR_CREATE_SCENARIOS);
+    }
+
+    public static DBHandler getDbHandler(Context context) {
+        if(dbHandler == null)
+            dbHandler = new DBHandler(context);
+
+        return dbHandler;
+    }
+
+    public Character addCharacter(Character newChar){
+        ContentValues charVals = new ContentValues();
+
+
+        charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_PLAYER_NAME, newChar.getPlayerName());
+        charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_CLASS_NAME, newChar.getClassName().toString());
+        charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_LEVEL, newChar.getLevel());
+        charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_EXP, newChar.getCurExp());
+        charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_MONEY, newChar.getMoney());
+        //charVals.put(PlayerContract.PlayerEntry.COLUMN_NAME_SCENARIO_INDEX, -1);
+
+        long charId = db.insert(PlayerContract.PlayerEntry.TABLE_NAME, null, charVals);
+
+        newChar.setId(charId);
+
+        return newChar;
+    }
+
+    public Scenario addScenario(Scenario newScen) {
+        ContentValues scenVals = new ContentValues();
+
+        scenVals.put(PlayerContract.ScenarioEntry.COLUMN_NAME_LEVEL, newScen.getLevel());
+        scenVals.put(PlayerContract.ScenarioEntry.COLUMN_NAME_HEALTH, newScen.getHealth());
+        scenVals.put(PlayerContract.ScenarioEntry.COLUMN_NAME_EXP, newScen.getExp());
+        scenVals.put(PlayerContract.ScenarioEntry.COLUMN_NAME_MONEY_TOKENS, newScen.getMoneyTokens());
+
+        long scenId = db.insert(PlayerContract.ScenarioEntry.TABLE_NAME, null, scenVals);
+
+        newScen.setId(scenId);
+
+        return newScen;
+    }
+
+    public ArrayList<Character> getAllCharacters(ArrayList<Character> characters) {
+        String charSelectQuery = "SELECT * FROM " + PlayerContract.PlayerEntry.TABLE_NAME;
+        String scenSelectQuery = "SELECT * FROM " + PlayerContract.ScenarioEntry.TABLE_NAME + " WHERE id=?";
+
+        Cursor c = db.rawQuery(charSelectQuery, null);
+
+        if (c.moveToFirst()) {
+            do {
+                //there's gotta be a better way to unpack a shitton of data like this. if this
+                //db had hundreds of columns I'd be fucked
+                Character newChar = new Character();
```

Comment:
just noting that this sets a default class, so something to potentially watch out for if this method changes and 
```
newChar.setClassName(c.getString(                    
c.getColumnIndex(PlayerContract.PlayerEntry.COLUMN_NAME_CLASS_NAME)));
```
is accidentally removed or something.